packages/arb-token-bridge-ui/src/components/TransferPanel/CustomFeeTokenUtils.ts

```diff
@@ -0,0 +1,51 @@
+import { Signer } from 'ethers'
+import { Provider } from '@ethersproject/providers'
+import { EthBridger, Erc20Bridger } from '@arbitrum/sdk'
+
+export async function approveCustomFeeTokenEstimateGas(params: {
+  erc20L1Address?: string
+  l1Signer: Signer
+  l1Provider: Provider
+  l2Provider: Provider
+}) {
+  const { erc20L1Address } = params
```

Comment:
No, I had to destructure `erc20L1Address` and then check it for `undefined` so that TypeScript can narrow it down. doing `if (typeof params.erc20L1Address !== 'undefined')` doesn't narrow it down.